migration/idempotent/017-metric-rollup.sql

```diff
@@ -122,19 +122,33 @@ BEGIN
         RETURN;
     END IF;
 
+    DROP TABLE IF EXISTS pending_registration;
+
+    -- Store metadata for newly created rollups which can be later used for registration in
+    -- prom_api.register_metric_view()
+    CREATE TEMPORARY TABLE pending_registration (
+        schema_name TEXT,
+        table_name TEXT,
+        resolution INTERVAL
+    );
+
     FOR r IN
         SELECT * FROM _prom_catalog.rollup
     LOOP
         rollup_view_created := FALSE;
         new_rollups_created := 0;
         FOR m IN
-            select id, metric_name, table_name FROM _prom_catalog.metric where id NOT IN (
+            select id, metric_name, table_name FROM _prom_catalog.metric where NOT is_view AND id NOT IN (
                 SELECT metric_id FROM _prom_catalog.metric_rollup WHERE rollup_id = r.id
             ) -- Get metric names that have a pending rollup creation.
```

Comment:
There is a unique index on `rollup_id, metric_id` so this ought to perform better:
```
select id, metric_name, table_name
from _prom_catalog.metric m
where not m.is_view
and not exists
(
    select 1
    from _prom_catalog.metric_rollup mr
    where mr.rollup_id = r.id
    and mr.metric_id = m.id
)
```